File: backend/components/metiers/threed/security/docs/samples/rbac/helpers/helpers_rbac.py
# helpers_rbac.py
# Helpers avancés pour la gestion des rôles et permissions dynamiques

def assign_role(user, role):
    """Simule l’assignation d’un rôle à un utilisateur"""
    user = dict(user)
    user['roles'] = user.get('roles', []) + [role]
    return user

def check_permission(user, permission):
    """Simule la vérification d’une permission"""
    return user and 'permissions' in user and permission in user['permissions']

File: backend/components/metiers/threed/security/docs/samples/rbac/helpers/helpers_rbac.js
// helpers_rbac.js
// Helpers avancés pour la gestion des rôles et permissions dynamiques

function assignRole(user, role) {
  // Simule l’assignation d’un rôle à un utilisateur
  return { ...user, roles: [...(user.roles || []), role] };
}

function checkPermission(user, permission) {
  // Simule la vérification d’une permission
  return user && user.permissions && user.permissions.includes(permission);
}

module.exports = { assignRole, checkPermission };
